### DISCUSSION TITLE
Extracting random structures from AIMD's vasprun.xml

### DISCUSSION THREAD

**Discussion by stemdkk** [ttadano/alamode#227] [Q&A]
Dear Dr. Tadano and users,
I am using displac.py to extract random displacement structures from AIMD. When I am simultaneously performing VASP6 on-the-fly machine learning during AIMD, the generated vasprun.xml produces the following error:The
range specified by --every is larger than the loaded MD data.   I checked and found that out of a total of 4000 steps, 126 steps are DFT-calculated structures, while the rest are predicted structures. When I change the range to 1:120:20, I can get results; however, using 1:130:20 gives an error. I suspect it might be an issue with the AIMD format, but I don't know how to check or modify it

**Comment by ttadano** [comment]
Hi,

Thank you for the report. Could you tell me how many snapshots were extracted when you run extract.py without `--every` option?

**Reply by stemdkk** [reply]
AIMD has a total of 4000 steps,--every=1001:3000:25, a total of 80 snapshots here. Thank you.

**Reply by ttadano** [reply]
Thank you. While I am not sure, the error occurred probably because the format of vasprun.xml differs when you use the on-the-fly machine-learning option.

**Reply by stemdkk** [reply]
Thank you for your reply. Because I have a lot of systems to calculate, how can I speed up my AIMD calculations on the basis of  accuracy?
